project/Build.scala

```diff
@@ -12,7 +12,7 @@ object DottyBuild extends Build {
   val baseVersion = "0.1"
   val isNightly = sys.env.get("NIGHTLYBUILD") == Some("yes")
 
-  val jenkinsMemLimit = List("-Xmx1300m")
+  val jenkinsMemLimit = List("-Xmx1500m")
```

Comment:
@felixmulder, I do understand that partest did the wrong thing. What I mean is that the current situation indicates that we use too much memory. Note, that even scalac runs entire test suite including partest, perfectly fine under `-Xmx1024m`(that's what is used by CI AFAIK).

The baselines, that seem reasonable to me are: 
 - normal hello-world like tests shouldn't need more than 100mb of memory.
 - dotty & stdlib tests shouldn't need more than 0.5 gigs of memory.

Assuming our CI runs 4 compiler threads, this means that in worst case, we would never need more than 1.2 Gb, and we'll only need 1.2 Gb if we're unlucky and dotty & stlib are compiled concurrently. If you'll have a look at the log output of the build failed with GC https://scala-ci.typesafe.com/job/dotty-master-validate-partest/2417/consoleFull, you'll see that these two tests _did not_ run concurrently.